app/scripts/metamask-controller.js

```diff
@@ -2060,6 +1955,35 @@ export default class MetamaskController extends EventEmitter {
       },
     );
 
+    this.controllerMessenger.subscribe(
+      'NotificationServicesPushController:onNewNotifications',
```

Comment:
Future note - I see other controllers drill down a `trackEvent` inside their controllers since the MetaMetrics controller is not shared across Mobile/Extension.

We can refactor this in the future, but currently we can utilise the events we emit (similar to what we did before)

Reply:
Sadly I didn't see a way where we can use the restricted messenger, this is not exposed from the init methods.

It would be nice to restrict this in the future.